From f176c9cc18c0ece3e408c50ab2953e367747700e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 21 Jun 2021 21:55:40 +0300
Subject: [PATCH] [SW-33447] gaudi: remove functions handling flr

Currently, pci device specific function such as setup, teardown,
enable and disable performs only FLR enable and disable. Since
FLR enable & disable functions are implemented as part of common
file, remove pci device specific function for gaudi.

Change-Id: I915a5582c485b2f88352c75a32eb9ad78634d7bf
---

diff --git a/arch/gaudi/common.c b/arch/gaudi/common.c
index 745079f..7d3b2d2 100644
--- a/arch/gaudi/common.c
+++ b/arch/gaudi/common.c
@@ -20,7 +20,6 @@
 #include <gic_regs.h>
 #include <nic0_phy_regs.h>
 
-#include <pcie_aux_masks.h>
 #include <pcie_dbi_masks.h>
 
 #include <hl_boot_if.h>			/* requires stdint.h */
@@ -404,31 +403,6 @@
 	preboot_puth((u32)loadaddr, false);
 
 	spi_copy((void *)loadaddr, offset, size, name);
-}
-
-/* Move FLR handling to S/W */
-void pci_dev_specific_enable(void)
-{
-	regmap_writel(0, mmPCIE_AUX_FLR_CTRL);
-}
-
-void pci_dev_specific_setup(void)
-{
-	pci_dev_specific_enable();
-}
-
-/* Move FLR handling to H/W */
-void pci_dev_specific_disable(void)
-{
-	u32 val = PCIE_AUX_FLR_CTRL_HW_CTRL_MASK |
-			PCIE_AUX_FLR_CTRL_INT_MASK_MASK;
-
-	regmap_writel(val, mmPCIE_AUX_FLR_CTRL);
-}
-
-void pci_dev_specific_teardown(void)
-{
-	pci_dev_specific_disable();
 }
 
 void pci_dev_specific_hot_rst(void)
